### ISSUE TITLE
Unhandled promise rejection error : Cannot read property 'statusCode' of undefined

### ISSUE THREAD

**Issue by prabushitha** [s-KaiNet/spsave#67]
When site's domain name resolution failure or any invalid site url, I get this error.
```
{
    "errorType": "Runtime.UnhandledPromiseRejection",
    "errorMessage": "TypeError: Cannot read property 'statusCode' of undefined",
    "reason": {
        "errorType": "TypeError",
        "errorMessage": "Cannot read property 'statusCode' of undefined"
    }
}
```

I tried handling the error in the catch statement as well as using a try catch, but both didn't catch the error.

```js
try {
  await spsave(coreOptions, credentials, fileOptions)
      .then((resp) => {
        logger.info('File uploaded to sharepoint');
        return Promise.resolve(resp);
      })
      .catch((err) => {
        logger.error('Catched error 1');
        return Promise.reject(err);
      })
} catch (e) {
    logger.error('Catched error 2', e);
}

``` 

Both error logs are not getting printed.

Server is getting crashed due to this.

**Comment by marc3218** [comment]
I just ran into the same issue. I've debugged the code with/within vs code. 
Let me share my results with you:

Error occurs in /lib/src/core/FileSaver.js in catch-block on line 95:

```javascript
.catch(function (err) {
    if (err && (err.response.statusCode === 500 || err.response.statusCode === 409)) { //line 95
        _this.tryReUpload(err, requestDeferred, attempts);
        return;
    }
    if (err && err.response.statusCode === 404 && err.response.body && err.response.body.toString().indexOf(FileSaver.directoryNotFoundCode) !== -1) {
        _this.foldersCreator.createFoldersHierarchy()
            .then(function () {
            _this.saveFile(requestDeferred, attempts + 1);
            return null;
        })
            .catch(function (folderError) {
            requestDeferred.reject(folderError);
        });
        return;
    }
    requestDeferred.reject(err);
  });
```

```err.response``` is undefined. 
If I delete everything inside the catch-block except ```requestDeferred.reject(err);```, then it works as expected for me

I hope one of the contributors can fix this.

**Comment by s-KaiNet** [comment]
Fixed in `4.0.1` version.
